src/main/java/cork/Engine.java

```diff
@@ -0,0 +1,170 @@
+    private static final String LOAD_COMMAND = "load";
+
+    private State state = State.HOME_SCREEN;
+    private Game currentGame;
+    private String gameName = "";
+    private UIHandler uiHandler = new UIHandler();
+
+    public static void main (String[] args) {
+        Engine engine = new Engine();
+        engine.run();
+    }
+
+    public void run()
+    {
+        while (state != State.QUIT)
+        {
+            switch (state)
+            {
+                case HOME_SCREEN:
+                    displayHomeScreen();
+                    break;
+                case MAIN_MENU:
+                    displayMainMenu();
+                    break;
+                case GAME_MENU:
+                    displayGameMenu();
+                    break;
+                case GAME_RUNNING:
+                    runGame();
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        uiHandler.exit();
+    }
+
+    private void changeState(final String choice)
+    {
+        if (choice.equalsIgnoreCase(QUIT_COMMAND)) {
+            state = State.QUIT;
+        } else {
+            switch (state)
+            {
+                case HOME_SCREEN:
+                    state = State.MAIN_MENU;
+                    break;
+                case MAIN_MENU:
+                    gameName = choice;
+                    state = State.GAME_MENU;
+                    break;
+                case GAME_MENU:
+                    if (choice.equals(UIHandler.LOAD_GAME_OPTION)) { loadGame(); }
+                    state = State.GAME_RUNNING;
+                    break;
+                case GAME_RUNNING:
+                    state = State.HOME_SCREEN;
+                    break;
+                default:
+                    state = State.QUIT;
+                    break;
+            }
+        }
+    }
+
+    private void displayHomeScreen()
+    {
+        changeState(uiHandler.displaySplashScreen());
+    }
+
+    private void displayMainMenu()
+    {
+        try { changeState(uiHandler.displayMainMenu(loadGameList())); }
+        catch (IOException e) {
+            state = State.QUIT;
+            uiHandler.displayError("Please ensure that the Cork JAR is in the same directory as the 'games' folder and that the folder is populated.");
+        }
+    }
+
+    private void displayGameMenu()
+    {
+        currentGame = new Game(gameName);
+        changeState(uiHandler.displayGameMenu(gameName));
+    }
+
+    private List<String> loadGameList() throws IOException {
+        try (Stream<Path> paths = Files.find(Paths.get(GAMES_DIRECTORY_PATH), 1, (path, attributes) -> attributes.isDirectory())) {
+            List<String> gameList = paths.map(Path::getFileName).map(Path::toString).collect(Collectors.toList());
+            gameList.remove(0);
+
+            if (gameList.isEmpty()) { throw new IOException("Games folder empty."); }
+
+            return gameList;
+        }

```

Comment:
The try block is never caught.
The try block is never followed by ( ); The code inside the brackets deserves its own try/catch block if handled locally or else the try block is redundant. If the try is removed, the possible 2 IOExeptions thrown by this method can be made into 2 custom exceptions. This would enable the caller to know specifically which one of them occurred.